mavlink-bindgen/src/parser.rs

```diff
@@ -710,6 +714,108 @@ impl MavEnumEntry {
             .map(|d| d.emit_tokens())
             .unwrap_or_default()
     }
+
+    #[inline(always)]
+    fn emit_params(&self) -> TokenStream {
+        if let Some(params) = &self.params {
+            let any_value_range = params.iter().any(|p| {
+                p.min_value.is_some()
+                    || p.max_value.is_some()
+                    || p.increment.is_some()
+                    || p.enum_used.is_some()
+                    || (p.reserved && p.default.is_some())
+            });
+            let any_units = params.iter().any(|p| p.units.is_some());
+            let lines = params
+                .iter()
+                .map(|param| param.emit_doc_row(any_value_range, any_units));
+            let mut table_header = "| Parameter | Description |".to_string();
+            let mut table_hl = "| --------- | ----------- |".to_string();
+            if any_value_range {
+                table_header += " Values |";
+                table_hl += " ------ |";
+            }
+            if any_units {
+                table_header += " Units |";
+                table_hl += " ----- |";
+            }
+            quote! {
+                #[doc = ""]
+                #[doc = "# Parameters"]
+                #[doc = ""]
+                #[doc = #table_header]
+                #[doc = #table_hl]
+                #(#lines)*
+            }
+        } else {
+            quote!()
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone, Default)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct MavParam {
+    pub index: usize,
+    pub description: Option<String>,
+    pub label: Option<String>,
+    pub units: Option<String>,
+    pub enum_used: Option<String>,
+    pub increment: Option<f32>,
+    pub min_value: Option<f32>,
+    pub max_value: Option<f32>,
+    pub reserved: bool,
+    pub default: Option<f32>,
+}
+
+impl MavParam {
+    fn format_valid_values(&self) -> String {
+        if self.reserved && self.default.is_some() {
+            format!("Reserved (use {})", self.default.unwrap())
```

Comment:
Clippy is not happy here (I also think it looks some annoying).

Can you try this?

```rs
if let (true, Some(default)) = (self.reserved, self.default)
```